src/Export/CacheHandler.php

```diff
@@ -0,0 +1,157 @@
+
+        return $this->cache->getItem($id);
+    }
+
+    private function getDynamicProductGroupOffsetCacheItem(int $offset): CacheItemInterface
+    {
+        $id = sprintf('%s_%s_%s', self::CACHE_ID_PRODUCT_GROUP, $this->shopkey, $offset);
+
+        return $this->cache->getItem($id);
+    }
+
+    private function setTotalInCache(CacheItemInterface $cacheItem, $total): void
+    {
+        $cacheItem->set($total);
+        $cacheItem->expiresAfter(self::CACHE_LIFETIME_PRODUCT_GROUP);
+        $this->cache->save($cacheItem);
+    }
+
+    private function getDynamicGroupsTotalCacheItem(): CacheItemInterface
+    {
+        $id = sprintf('%s_%s_total', self::CACHE_ID_PRODUCT_GROUP, $this->shopkey);
+
+        return $this->cache->getItem($id);
+    }
+
+    /**
+     * Gets the total count of dynamic product groups from cache.
+     */
+    private function getDynamicProductGroupTotalFromCache(): int
+    {
+        $cacheItem = $this->getDynamicGroupsTotalCacheItem();
+        if ($cacheItem && $cacheItem->isHit()) {
+            $cacheItem->expiresAfter(self::CACHE_LIFETIME_PRODUCT_GROUP);
+            $this->cache->save($cacheItem);
+
+            return (int)$cacheItem->get();
+        }
+
+        return 0;
+    }
+
+    public function isDynamicProductGroupTotalCached(): bool
+    {
+        $totalCacheItem = $this->getDynamicGroupsTotalCacheItem();
+
+        return $totalCacheItem && $totalCacheItem->isHit();
+    }
+
+    public function setDynamicProductGroupTotal(int $total): void
+    {
+        $totalCacheItem = $this->getDynamicGroupsTotalCacheItem();
+        $this->setTotalInCache($totalCacheItem, $total);
+    }
+
+    public function setDynamicProductGroupOffset($products, int $offset): void
+    {
+        if ($offset > 0) {
+            $cacheItem = $this->getDynamicProductGroupOffsetCacheItem($offset);
+            $cacheItem->set($products);
+            $cacheItem->expiresAfter(self::CACHE_LIFETIME_PRODUCT_GROUP);
+            $this->cache->save($cacheItem);
+        }
+    }
+
+    public function isCacheWarmedUp(int $offset): bool
+    {
+        $cacheItem = $this->getDynamicProductGroupOffsetCacheItem($offset);
+        if ($cacheItem && $cacheItem->isHit()) {
+            $cacheItem->set(true);
+            $cacheItem->expiresAfter(self::CACHE_LIFETIME_PRODUCT_GROUP);
+            $this->cache->save($cacheItem);
+
+            return true;
+        }
+
+        return false;
+    }
+
+    public function isDynamicProductGroupsCached(): bool
+    {
+        $cacheItem = $this->getDynamicProductGroupWarmedUpCacheItem();
+        if ($cacheItem && $cacheItem->isHit()) {
+            $cacheItem->set(true);
+            $cacheItem->expiresAfter(self::CACHE_LIFETIME_PRODUCT_GROUP);
+            $this->cache->save($cacheItem);
+
+            return true;
+        }
+
+        return false;
+    }
+
+    public function getCachedCategoryIds(int $offset): array
```

Comment:
Some methods require an offset, while others don't and there is no clear distinction between why. Make the purpose of the method more clear, by giving it a more explicit name.